### ISSUE TITLE
Rooms Switch bug

### ISSUE THREAD

**Issue by netRunner0** [nicoh88/homebridge-xiaomi-roborock-vacuum#119]
Accessories are divided, but when you move one to the desired room - everything moves!)
![IMG_0541](https://user-images.githubusercontent.com/44049295/75430390-0f3a2e80-596d-11ea-945a-2c85bbe70af2.jpeg)

**Comment by c-o-m-m-a-n-d-e-r** [comment]
they are not really divided. these are all subservices.
not really a bug. i think its a design mistake.

we have to think about the design i think.

**Comment by c-o-m-m-a-n-d-e-r** [comment]
ok looking deeper in it, i think there is a bug with serialnumber etc... will try to fix this!

**Comment by c-o-m-m-a-n-d-e-r** [comment]
ok if i'm not totally wrong, then this would be only possible when you create  an accessory for each room ... mhhh thats not what we want.

i will search for a solution, but i think this won't work

**Comment by netRunner0** [comment]
Also, if you turn on the cleaning in the room, the main switch for the general cleaning remains off

**Comment by netRunner0** [comment]
![IMG_0542](https://user-images.githubusercontent.com/44049295/75450765-373b8900-5991-11ea-81ca-eddc367d47aa.jpeg)

**Comment by netRunner0** [comment]
With the latest update, the status of the main switch changes, when the zone cleaning starts,
however, after a while both are turned off, although the robot continues to clean.

**Comment by netRunner0** [comment]
With the latest updates, when you turn on the cleaning in the room, all room switches become active:


[4/13/2020, 06:38:29] [Roborock S6] INF getBatteryLow | roborock.vacuum.s6 | Batterylevel is 94%
[4/13/2020, 06:38:29] [Roborock S6] INF getCleaning | roborock.vacuum.s6 | Cleaning is true
[4/13/2020, 06:38:29] [Roborock S6] INF getSpeed | roborock.vacuum.s6 | Fanspeed is 102 over miIO. Converting to HomeKit
[4/13/2020, 06:38:29] [Roborock S6] INF getSpeed | roborock.vacuum.s6 | Fanspeed is 102 over miIO "Balanced" > HomeKit speed 60%
[4/13/2020, 06:38:29] [Roborock S6] INF getCleaning | roborock.vacuum.s6 | Cleaning is true
[4/13/2020, 06:38:29] [Roborock S6] INF getDocked | roborock.vacuum.s6 | Robot Docked is false (Status is room-cleaning)
[4/13/2020, 06:38:29] [Roborock S6] INF getCleaning | roborock.vacuum.s6 | Cleaning is true
[4/13/2020, 06:38:29] [Roborock S6] INF getPauseState | roborock.vacuum.s6 | Pause possible is true
[4/13/2020, 06:38:29] [Roborock S6] INF getCleaning | roborock.vacuum.s6 | Cleaning is true

**Comment by afharo** [comment]
Yes! This is an issue that we need to fix. Any help is needed! Thank you for sharing the logs!
Maybe @c-o-m-m-a-n-d-e-r  can take a look at it?

**Comment by c-o-m-m-a-n-d-e-r** [comment]
yeahr ... but don't have enough spare time at the moment ... i will try to look at it ... any other help is very welcome

**Comment by netRunner0** [comment]
But how to add a room to the current cleaning without completing the current cleaning?

**Comment by afharo** [comment]
Is that possible via the original app?
I tried in the Xiaomi Mi app and even editing rooms can’t be done until the cleaning is complete.
